modules/post/windows/manage/forward_pageant.rb

```diff
@@ -40,103 +45,112 @@ def initialize(info = {})
         }
       )
     )
-    register_options(
-      [
-        OptString.new('SocketPath', [false, 'Specify a filename for the local UNIX socket.', nil])
-      ]
-    )
+    register_options([
+      OptString.new('SocketPath', [false, 'Specify a filename for the local UNIX socket.', nil])
+    ])
+  end
+
+  def sockpath
+    @sockpath ||= "#{Dir.tmpdir}/#{Rex::Text.rand_text_alphanumeric(8)}"
```

Comment:
Using `Dir.tmpdir` is potentially racey. After opening, we immediately call `chmod 0700` on the file path, which *should* have minimal impact in the unlikely event of exploitation, although a symlink to SSH keys or sudoers file would result in denial of service, as `600` permissions are required.

Other module also use `tmpdir` :

```
# grep -rn "tmpdir" modules/ | grep require
modules/post/multi/gather/firefox_creds.rb:9:require 'tmpdir'
modules/auxiliary/server/tftp.rb:6:require 'tmpdir'
```

(I thought I also found one instance of a hard-coded `"/tmp/"` local file path, but can't seem to find it now.)

Unfortunately `::UNIXServer.open()` requires providing a file name as input. It does not appear to allow file creation upon initialization (`listen` does, but is reserved for TCP, not UNIX sockets).

We can't use our traditional temporary file creation method (`Rex::Quickfile`) as `::UNIXServer.open()` requires the file to not exist and requires a file path not a file handle.